tests/apollo/util/bft_debug_tool.py

```diff
@@ -0,0 +1,140 @@
+# Concord
+#
+# Copyright (c) 2022 VMware, Inc. All Rights Reserved.
+#
+# This product is licensed to you under the Apache 2.0 license (the "License").
+# You may not use this product except in compliance with the Apache 2.0 License.
+#
+# This product may include a number of subcomponents with separate copyright
+# notices and license terms. Your use of these subcomponents is subject to the
+# terms and conditions of the subcomponent's license, as noted in the LICENSE
+# file.
+
+#from util.bft import BftTestNetwork
+
+import os
+import psutil
+import shutil
+import subprocess
+from pathlib import Path
+
+class BftDebugTool:
+    """
+    This class handles all operations that should be done to support debug and analysis tools such as Heaptrack/ASAN/TSAN etc.
+    """
+    sanitizer_names = ["TSAN", "ASAN", "UBSAN"]
+
+    def __init__(self, num_replicas, builddir, current_test):
+        """
+        Initialize debug tool name and its output folder, path and prefixes for the current test case
+        """
+        self.external_tool_procs = {}   # process info for external tools (e.g Valgrind and Heaptrack)
+        self.name = None                # None if there is no debug tool, else upper case name of current test debug tool
+        self.output_log_path = None
+        self.output_log_prefix = None
+        output_folder_name = None
+        name = None
+        self.output_counters = {i: 0 for i in range(num_replicas)} # used to count output files per specific replica in the same test case
+        if 'UBSAN_OPTIONS' in os.environ:
```

Comment:
@samcramer  that pratiaclly can't happen if you look at how CMakefile generates tests.
It can happen if user runs a test manually (with a very long text command) with all enviorment variables given from command line.
So I agree, it would be better to check that only one is uo here.
Thanks.